include/zephyr/kernel/thread.h

```diff
@@ -372,6 +372,10 @@ struct k_thread {
 	_wait_q_t  halt_queue;
 #endif /* CONFIG_SMP */
 
+#ifdef CONFIG_KERNEL_WARN_PENDING
+	int warn_pending_seconds;
+#endif /* CONFIG_KERNEL_WARN_PENDING */
```

Comment:
I added a variable here to indicate the desired behavior of each thread, because considering that some threads may not need these optimizations, there are some warnings that may be misleading, so I made this function depend on each thread and depends on the this variable of the thread.

Reply:
Such usage may require adding a Public API to perform **GET** **SET**, so I need to confirm first that this method is acceptable to everyone?

Reply:
<img width="1407" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/26378606/921e5b29-0789-491c-bc72-341fdaa5ed3f">